.github/workflows/notebooks.yml

```
@@ -0,0 +1,28 @@
+name: XGI Notebooks
+
+on: [push, pull_request]
```

```suggestion
on:
  push:
    branches:
    - main
  pull_request:
    branches:
    - main
```

Something like this will also prevent from github actions running twice in every PR.

---

Good idea, thanks!